src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -201,7 +203,9 @@ static void ScanStackRoots(Thread * pThread, promote_func* fn, ScanContext* sc)
 #if defined(FEATURE_EH_FUNCLETS)
         flagsStackWalk |= GC_FUNCLET_REFERENCE_REPORTING;
 #endif // defined(FEATURE_EH_FUNCLETS)
+        gcctx.pScannedSlots = new SetSHash<Object**, PtrSetSHashTraits<Object**> >();
```

Comment:
I'd prefer to see this not allocate at all unless duplicate detection is turned on. The performance impact doesn't really concern me, but it's just an extra opportunity for a process that is nearly out of memory to get an unlucky OOM in a new and horrible place, as I don't know think flowing a threw this code during the middle of a GC will leave the GC in a consistent state. Alternatively, you could do a nothrow'ing new, and have a null check around the use of the pScannedSlot variable.

Reply:
Sigh, my mistake, that was not intended. Happened accidentally by transforming the previous state of the PR to the current one.

Reply:
Fixed in the last commit